monocle/gitlab/graphql.py

```diff
@@ -0,0 +1,166 @@
+# Monocle.
+# Copyright (C) 2019-2020 Monocle authors
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import logging
+from time import sleep
+
+import requests
+from monocle import utils
+from tenacity import (
+    after_log,
+    retry,
+    retry_if_exception_type,
+    stop_after_attempt,
+    wait_fixed,
+)
+
+
+class RequestTimeout(Exception):
+    pass
+
+
+class RequestException(Exception):
+    pass
+
+
+class GitlabGraphQLQuery(object):
+
+    log = logging.getLogger(__name__)
+
+    def __init__(self, token_getter):
+        self.url = "{github_url}/api/graphql"
```

Comment:
not sure where this is interpolated, so url name might be different
```suggestion
        self.url = "{gitlab_url}/api/graphql"
```

Reply:
good catch. I think url is not required. Removed it.